src/tcg-to-deckbox.py

```diff
@@ -1,21 +1,77 @@
 import sys
 import csv
+import time
 import os
+import os.path
 import configparser
 import re
 from tkinter import filedialog
 from tkinter import messagebox
 import tkinter as tk
+import requests
+import urllib.parse
+import ssl
+import json
+
+
+# Constants
+MULTI_NAMES_FILE = "multiple_names.json"
+SCRYFALL_URL = "https://api.scryfall.com/cards/search?order=cmc&q=%28is%3Adoublesided%20OR%20is%3Asplit%20OR%20is%3Aadventure%29%20AND%20game%3Apaper%20AND%20-is%3Atoken%20AND%20-set%3ACMB1%20AND%20-is%3Aextra"
+DECKBOX_URL = "https://deckbox.org/mtg/"
+
+# Global Data
+scryfall_data = {}
+
+# Global replacement helpers
+ixalan_bab = [
+    "Legion's Landing",
+    "Search for Azcanta",
+    "Arguel's Blood Fast",
+    "Vance's Blasting Cannons",
+    "Growing Rites of Itlimoc",
+    "Conqueror's Galleon",
+    "Dowsing Dagger",
+    "Primal Amulet",
+    "Thaumatic Compass",
+    "Treasure Map",
+]
+
+bab_mapping = {
+    "Impervious Greatwurm": "Guilds of Ravnica",
+    "Kenrith, the Returned King": "Throne of Eldraine",
+    "Realmwalker": "Kaldheim",
+    "Vorpal Sword": "Adventures in the Forgotten Realms",
+}
 
 # Get rid of the root TK window, we don't need it.
 root = tk.Tk()
 root.withdraw()
 
+# Queries scryfall to build a list of cards that have multiple names.
+def fetch_multiple_names(uri, page=1):
+    print(
+        "Begin: Download '%s', page %s of results" % (urllib.parse.unquote(uri), page)
+    )
+    try:
+        with requests.get(uri) as scryfall_response:
```

Comment:
if we track the etag header (append to the file name or add it as a top level key in the file), and pass it back in the "If-None-Match" header on the GET request, the api will respond with 304 Not Modified if the resulting content hasn't updated on their end.  If not modified we'll be able to short circuit refreshing the scryfall data.

Reply:
Makes sense, I can look into this.